sdk/python/pulumi_azure/ad/application.py

```diff
@@ -8,6 +8,7 @@
 import pulumi.runtime
 from .. import utilities, tables
 
+warnings.warn("The azure/ad module has moved to a dedicated Pulumi azuread package", DeprecationWarning)
```

Comment:
i'm concerned about thsi.  say someone imports `azure`.  Doesn't `azure` import `ad`?  Won't that cause this path to run and warn just on import?  SHouldn't we just do this inside the constructor?

Reply:
We also have the pulumi.log.warn in the constructor:

```
pulumi.log.warn("application is deprecated: The azure/ad module has moved to a dedicated Pulumi azuread package")
```

The warnings.warn is just the Python-esque way of decorating like `// @deprecated...` in TypeScript

Reply:
have you tested to make sure this isn't seen when someone uses python?  and only seen if they actually create the resource?

Reply:
yes - I have tested both of them - you only get warnings.warn if you use the python interpretor directly

FWIW, the same things are used within classes as well

```
warnings.warn("explicit use of __name__ is deprecated", DeprecationWarning)
```

Reply:
> yes - I have tested both of them

Ok.  Then i'm ok :)   Just trying to be on the safe side here.